src/User/CacheRepository.php

```diff
@@ -4,31 +4,46 @@
 
 namespace CultuurNet\UDB3\User;
 
+use DateTimeImmutable;
 use Doctrine\Common\Cache\Cache;
 
 class CacheRepository implements Auth0ManagementTokenRepository
 {
-    /**
-     * @var Cache
-     */
-    private $cache;
+    private const TOKEN_KEY = 'auth0_token';
 
+    private Cache $cache;
 
     public function __construct(Cache $cache)
     {
         $this->cache = $cache;
     }
 
-    public function token(): ?string
+    public function token(): ?Auth0Token
     {
-        if (!$this->cache->contains('token')) {
+        if (!$this->cache->contains(self::TOKEN_KEY)) {
             return null;
         }
-        return $this->cache->fetch('token');
+
+        $tokenAsArray = json_decode($this->cache->fetch(self::TOKEN_KEY), true);
```

Comment:
Nit: The token that's currently in the cache (if any) when we deploy this will just be a string, so maybe we need to check if `$tokenAsArray` is false and if so also return `null`.

Reply:
To solve that I changed the key value for Redis. So we create a new key/value pair the first time it gets deployed.

Reply:
Perfect 👍